R/row_to_names.R

```diff
@@ -1,21 +1,48 @@
 #' Elevate a row to be the column names of a data.frame.
 #'
 #' @param dat The input data.frame
-#' @param row_number The row of \code{dat} containing the variable names
-#' @param remove_row Should the row \code{row_number} be removed from the resulting data.frame?
-#' @param remove_rows_above If \code{row_number != 1}, should the rows above \code{row_number} - that is, between
-#'   \code{1:(row_number-1)} - be removed from the resulting data.frame?
+#' @param row_number The row of \code{dat} containing the variable names or the
+#'   string \code{"find_header"} to use \code{find_header(dat=dat, ...)} to find
+#'   the row_number.
+#' @param ... Sent to \code{find_header()}, if
+#'   \code{row_number = "find_header"}.  Otherwise, ignored.
+#' @param remove_row Should the row \code{row_number} be removed from the
+#'   resulting data.frame?
+#' @param remove_rows_above If \code{row_number != 1}, should the rows above
+#'   \code{row_number} - that is, between \code{1:(row_number-1)} - be removed
+#'   from the resulting data.frame?
 #' @return A data.frame with new names (and some rows removed, if specified)
+#' @family Set names
 #' @examples
 #' x <- data.frame(X_1 = c(NA, "Title", 1:3),
 #'                 X_2 = c(NA, "Title2", 4:6))
 #' x %>%
 #'   row_to_names(row_number = 2)
+#'
+#' x %>%
+#'   row_to_names(row_number = "find_header")
 #' @export
-row_to_names <- function(dat, row_number, remove_row = TRUE, remove_rows_above = TRUE) {
+row_to_names <- function(dat, row_number, ..., remove_row = TRUE, remove_rows_above = TRUE) {
   # Check inputs
-  if (length(row_number) != 1 | !is.numeric(row_number)) {
-    stop("row_number must be a numeric of length 1")
+  if (!is.logical(remove_row) & length(remove_row) == 1) {
+    stop("remove_row must be a logical scalar, not ", as.character(remove_row))
```

Comment:
Would it be clearer to people who don't know what a scalar is to say "`must be either TRUE or FALSE`?"  Also applies to subsequent use of "logical scalar"